### PR TITLE
Issue #934 - Make Eval covariant

### PR DESCRIPTION
**Pull request by alexandru**
The reason is that `Eval` should be a drop-in replacement for by-name parameters. Noticed that changing it wouldn't break anything within Cats.

### CODE REVIEW

File: core/src/main/scala/cats/Eval.scala [hunk 1]
```diff
@@ -350,7 +350,7 @@ trait EvalMonoid[A] extends Monoid[Eval[A]] with EvalSemigroup[A] {
 trait EvalGroup[A] extends Group[Eval[A]] with EvalMonoid[A] {
   implicit def algebra: Group[A]
   def inverse(lx: Eval[A]): Eval[A] =
-    lx.map(_.inverse)
+    lx.map(_.inverse())
```
Comment (adelbertc) [inline core/src/main/scala/cats/Eval.scala]: Why is this needed?

Comment (alexandru) [inline core/src/main/scala/cats/Eval.scala]: It isn't needed, but the definition of that function in `cats.syntax.GroupOps` is:

```scala
def inverse(): A = macro Ops.unop[A]
```

Not sure why that's the definition, but when you've got a parens pair like that in your `def`, then you have tools like IntelliJ IDEA complaining about you not using the parens at the call site. Which I personally find cool because those parens usually indicate side-effects. Maybe we should change the `def inverse()` definition?

Comment (ceedubs) [inline core/src/main/scala/cats/Eval.scala]: I tried removing the parens from `def inverse()` and got this:

```
[error] /Users/cody/code/cats/core/src/main/scala/cats/syntax/group.scala:15: macro implementation has incompatible shape:
[error]  required: (c: scala.reflect.macros.whitebox.Context): c.Expr[A]
[error]  or      : (c: scala.reflect.macros.whitebox.Context): c.Tree
[error]  found   : (c: reflect.macros.Context)(): c.Expr[R]
[error] number of parameter sections differ
[error]   def inverse: A = macro Ops.unop[A]
```

Paging @non, our macro optimization wizard.

Comment (milessabin) [inline core/src/main/scala/cats/Eval.scala]: You would also need to remove the parens from the corresponding macro defn.

**Comment by ceedubs** [comment]
It's been over 2 weeks since https://github.com/typelevel/cats/issues/934#issuecomment-197871798 and nobody has spoken up against this. :+1: from me.

**Comment by non** [comment]
I agree with the rationale. Let's do it! 👍
